app/models/talentlms/facade.rb

```diff
@@ -10,42 +10,116 @@ def initialize
       @api = Config.first
     end
 
+    private def lms_find_user_by_id(id)
+      result = @api.post('users', { id: id })
+      result
+    rescue RuntimeError => e
+      raise e unless e.message.include?('The requested user does not exist')
+    end
+
+    private def lms_find_user_by_email(email)
+      result = @api.post('users', { email: email })
+      result
+    rescue RuntimeError => e
+      raise e unless e.message.include?('The requested user does not exist')
+    end
+
+    private def lms_find_user_by_username(username)
+      result = @api.post('users', { username: username })
+      result
+    rescue RuntimeError => e
+      raise e unless e.message.include?('The requested user does not exist')
+    end
+
+    # Login username to TalentLMS
+    #
+    # ENV['DEV_OFFSET'] can be set to an integer to prevent username collisions between development environments
+    #
+    # @@param user [User] the user
+    # @return [String] username to be used for generated users in TalentLMS
+    def lms_username(user)
+      "#{ENV['RAILS_ENV']}_#{user.id + Integer(ENV.fetch('DEV_OFFSET', 0))}"
```

Comment:
Is the integer useful in production environments?  If not, maybe just:
```suggestion
      username = "#{ENV['RAILS_ENV']}_#{user.id}"
      return username unless Rails.env.development?
      
      "#{username}_#{ENV.fetch('DEV_OFFSET', 0)}"
```

I think what you have will increment user ids rather than adding an increment at the end, which might be the intention, but seems odd, it would mean your user_id 1 is my user_id 2 potentially.

Reply:
This makes sense and definitely feels cleaner. I was using the existing logic, just pulled into its own method to prevent duplication. I'll push this change up.